### ISSUE TITLE
[SPACES] List&Browse Spaces content

### ISSUE THREAD

**Issue by jesmrec** [owncloud/android#3807]
To do after #3803 and #3805

By clicking an space from the list, the following items should be visible:

  - List of available files/folders in the space. 
  - Item in the bottom bar to open the list of spaces (already added in #3803, but now, action to browse added)
  - First item in the bottom bar to be named `Personal`. Only for spaces accounts, for non-spaces accounts it stays as `Files` (already added in #3803)
  - Between the top bar and the list of files, a layout with the image, name and subtitle of the space. Only in space's root folder, not in the non-root, neither in personal space.
  - On the top bar, display the space name instead of app name
  - Optional: description (no markdown support ftm)

Browsing forth and back through spaces folder tree  available as well. Using the webdavurl from graph API's drives endpoint. For non-root folders, showing the folder name. Regular browsing.

Nice to have: save breadcrumbs (browsing in the upper bar like iOS, as example).

### TASKS

- [x] Research (if needed)
- [x] Create branch spaces/list_content
- [x] Development tasks
  - [x] Add click action to every item in the spaces list to retrieve list of files
  - [x] Use upsert instead of insertOrReplace when fetching spaces
  - [x] Add layout to display space info in the space root folder
  - [x] Change to space name in the top bar and add up arrow instead of drawer
  - [x] Browsing through (maybe nothing to do here)
- [x] Code review and apply changes requested
- [x] Design test plan
- [x] QA
- [x] Merge branch spaces/list_content into spaces/main
